openqaoa/problems/converters.py

```diff
@@ -0,0 +1,384 @@
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import numpy as np
+from collections import defaultdict
+from .problem import QUBO
+from ..optimizers.result import Result
+
+
+class FromDocplex2IsingModel:
+    def __init__(self, model, multipliers: float = None):
+
+        """
+        Creates an instance to translate Docplex models to its Ising Model representation
+
+        Parameters
+        ----------
+        model : Docplex model
+            It is an object that has the mathematical expressions (Cost function,
+            Inequality and Equality constraints) of an optimization problem in
+            binary representation.
+
+        """
+        # assign the docplex Model
+        self.model = model.copy()
+
+        # save the index in a dict
+        self.idx_terms = {}
+        for x in self.model.iter_variables():
+            self.idx_terms[x] = x.index
+            if x.vartype.short_name != "binary":
+                TypeError(f"Variable {x.vartype.short_name} is not allowed.")
+        # get doclex qubo and ising model
+        self.qubo_docplex, self.ising_model = self.get_models(multipliers)
+
+    def linear_expr(self, expr):
+        """
+        Adds linear expresions to the objective function stored in self.qubo_dict.
+
+        Parameters
+        ----------
+        expr : model.objective_expr
+            A docplex model attribute.
+
+        """
+        for x, weight in expr.get_linear_part().iter_terms():
+            self.qubo_dict[(self.idx_terms[x],)] += weight
+
+    def quadratic_expr(self, expr):
+        """
+        Adds quadratic terms to the objective function stored in self.qubo_dict.
+
+        Parameters
+        ----------
+        expr : model.objective_expr
+            A docplex model attribute.
+        """
+
+        # save the terms and coeffs of the quadratic part,
+        # considering two index i and j
+        for x, y, weight in expr.iter_quad_triplets():
+            i = self.idx_terms[x]
+            j = self.idx_terms[y]
+            if i == j:  # if this is the term is  Z1**2 for example
+                self.qubo_dict[(i,)] += weight
+            else:
+                self.qubo_dict[(i, j)] += weight
+
+    def Equality2Penalty(self, expression, multiplier: float):
+        """
+        Add equality constraints to the cost function using the penality representation.
+        The constraints should be linear.
+        
+        Parameters
+        ----------
+        expression : docplex.mp.linear.LinearExpr
+            docplex equality constraint "expression == 0".
+        multiplier : float
+            Lagrange multiplier of the penalty
+
+        Returns
+        -------
+        penalty : docplex.mp.quad.QuadExpr
+            Penalty that will be added to the cost function.
+
+        """
+        penalty = multiplier * (expression) ** 2
+        return penalty
+
+    def bounds(self, linear_expression):
```

Comment:
This method also seems like it could be a staticmethod

Reply:
sure, I'll fix it.